### DISCUSSION TITLE
ImGUI OpenGL3 Linker errors and "Failed to initialize OpenGL loader!" errors

### DISCUSSION THREAD

**Discussion by CrispulentCrisps** [ocornut/imgui#6949] [New Users Build/Link/Run issues ONLY!]
Hello, I'm new to openGL and have been trying to convert my program from an SDL renderer to openGL for a performance boost, as otherwise the program I am making runs at about 15.7FPS

With trying to link the opengl it is fine for the backend implementation, asides from the "Failed to initialize OpenGL loader!" error

I had tried to work glfw, but that was getting linker errors as well with the glfw backend implementation
`
if (!ImGui_ImplOpenGL3_Init("#version 300")) {
	cout << glGetError();
}
`
the cout line is throwing a linker error

any help is appreciated!

**Comment by ocornut** [comment]
Which linker error ?

**Reply by CrispulentCrisps** [reply]
`Error	LNK2019	unresolved external symbol __imp_glGetError referenced in function "public: void __cdecl Tracker::Run(void)"`

This is the exact error

**Reply by ocornut** [reply]
You need to need with the opengl library. Opengl32.lib on windows.

**Reply by CrispulentCrisps** [reply]
would this be within the imgui folder itself or elsewhere?

**Reply by CrispulentCrisps** [reply]
also do you mean a seperate library like GLEW?
and if so then for a music creation program what would be the best option?
